src/pyFAI/multi_geometry.py

```diff
@@ -342,3 +343,493 @@ def reset(self, collect_garbage=True):
+        :param lst_mask: numpy.Array or list of numpy.array which mask the lst_data.
+        :param float dummy: value for dead/masked pixels
+        :param float delta_dummy: precision for dummy value
+        :param lst_variance: list of array containing the variance of the data. If not available, no error propagation is done
+        :type lst_variance: list of ndarray
+        :param float polarization_factor: polarization factor between -1 (vertical) and +1 (horizontal).
+                * 0 for circular polarization or random,
+                * None for no correction,
+                * True for using the former correction
+        :param ndarray dark: dark noise image
+        :param lst_flat: numpy.Array or list of numpy.array which flat the lst_data.
+        :param IntegrationMethod method: IntegrationMethod instance or 3-tuple with (splitting, algorithm, implementation)
+        :param float normalization_factor: Value of a normalization monitor
+        :return: regrouped intensity and unit arrays
+        :rtype: Integrate2dResult
+        """
+        if "npt_horizontal" in kwargs:
+            deprecated_warning(type_=type(kwargs["npt_horizontal"]), name="npt_horizontal", replacement="npt_ip", since_version="2024.11/12")
+            npt_ip = kwargs["npt_horizontal"]
+        if "npt_vertical" in kwargs:
+            deprecated_warning(type_=type(kwargs["npt_vertical"]), name="npt_vertical", replacement="npt_oop", since_version="2024.11/12")
+            npt_oop = kwargs["npt_vertical"]
+        if "horizontal_unit" in kwargs:
+            deprecated_warning(type_=type(kwargs["horizontal_unit"]), name="horizontal_unit", replacement="unit_ip", since_version="2024.11/12")
+            unit_ip = kwargs["horizontal_unit"]
+        if "vertical_unit" in kwargs:
+            deprecated_warning(type_=type(kwargs["vertical_unit"]), name="vertical_unit", replacement="unit_oop", since_version="2024.11/12")
+            unit_oop = kwargs["vertical_unit"]
+        if "horizontal_unit_range" in kwargs:
+            deprecated_warning(type_=type(kwargs["horizontal_unit_range"]), name="horizontal_unit_range", replacement="ip_range", since_version="2024.11/12")
+            ip_range = kwargs["horizontal_unit_range"]
+        if "vertical_unit_range" in kwargs:
+            deprecated_warning(type_=type(kwargs["vertical_unit_range"]), name="vertical_unit_range", replacement="oop_range", since_version="2024.11/12")
+            oop_range = kwargs["vertical_unit_range"]
+
+        for fi in self.ais:
+            unit_ip, unit_oop = fi.parse_units(unit_ip=unit_ip, unit_oop=unit_oop,
+                                             sample_orientation=sample_orientation)
+
+            fi.reset_integrator(incident_angle=unit_ip.incident_angle,
+                                tilt_angle=unit_ip.tilt_angle,
+                                sample_orientation=unit_ip.sample_orientation)
+
```

Comment:
Always convert method to an instance of IntegrationMethod and then check the attributes.